pkg/web/counterparties.go

```diff
@@ -157,6 +160,17 @@ func (s *Server) CreateCounterparty(c *gin.Context) {
 	// The source of the counterparty created by the API is user entry
 	counterparty.Source = enum.SourceUserEntry
 
+	// Parse and ensure the website has a protocol (default to `https://`)
+	if counterparty.Website.Valid {
+		if website, err := counterparty.NormalizedWebsite(); err != nil {
+			c.Error(err)
+			c.JSON(http.StatusBadRequest, api.Error("cannot parse counterparty website string"))
+			return
+		} else {
+			counterparty.Website.String = website
+		}
+	}
```

Comment:
Website is not required for an API request so we don't have to return a 400 here unless it is not parseable.

```suggestion
	if counterparty.Website.String, _ = counterparty.NormalizedWebsite(); err != nil {
		if !errors.Is(errors.ErrNullString) {
			c.Error(err)
			c.JSON(http.StatusBadRequest, api.Error("invalid website url")
			return
		}
	}
```